transform/mattermost-analytics/models/intermediate/product/licenses/_licenses_servers/int_cloud_licenses.sql

```diff
@@ -0,0 +1,11 @@
+select s.cws_installation as installation_id
+        , c.portal_customer_id as customer_id
+        , c.email as email
+        , c.name as company_name
+        , coalesce(s.edition, p.name) as license_name
+        , 'Stripe' as source
+    from {{ ref('stg_stripe__subscriptions')}}  s 
+    join {{ ref('stg_stripe__customers')}}  c on s.customer_id = c.customer_id
+    left join {{ ref('stg_stripe__products')}} p on coalesce(s.product_id, s.current_product_id) = p.product_id
+    where s.cws_installation is not null
+    qualify row_number() over (partition by cws_installation order by s.edition desc nulls last) = 1
```

Comment:
`installation_id` of null is not useful since it doesn't provide any information about the license, and there seems to exist another row with the actual license name. Overall design is not effected, only the data that is output.

Reply:
`installation_id` is not null.`edition` might be null.  This results in four possible sub-cases: (a) single row with non null edition, (b) single row with null edition, (c) multiple rows with different non null edition and (d) multiple rows with a mix of non null edition and null values.

Cases (a) and (b) should be OK. 
Case (c) is handled by alphabetically sorting by edition and picking the first. Why does this address the duplication? 
Case (d) is a variant of case (c).

Reply:
You are right about Case(c). I am looking into it. Thanks

Reply:
I updated the qualify to use `current_period_end_at` which will fetch the latest row for the installation. I checked a few instances and its working as expected.